urllib3/connectionpool.py

```diff
@@ -396,7 +396,7 @@ def _make_request(self, conn, method, url, timeout=_Default, chunked=False,
 
         try:
             assert_header_parsing(httplib_response.msg)
-        except HeaderParsingError as hpe:  # Platform-specific: Python 3
+        except (HeaderParsingError, TypeError) as hpe:  # Platform-specific: Python 3
```

Comment:
Hrm. Take a look at `assert_header_parsing`: can we enhance that to avoid throwing the `TypeError`?